common/src/main/java/no/difi/meldingsutveksling/CertificateParser.java

```diff
@@ -1,25 +1,28 @@
-package no.difi.meldingsutveksling.noarkexchange;
+package no.difi.meldingsutveksling;
 
 import org.bouncycastle.cert.X509CertificateHolder;
 import org.bouncycastle.cert.jcajce.JcaX509CertificateConverter;
-import org.bouncycastle.jcajce.provider.asymmetric.x509.CertificateFactory;
 import org.bouncycastle.openssl.PEMParser;
 
 import java.io.IOException;
+import java.io.Reader;
 import java.io.StringReader;
-import java.security.cert.Certificate;
 import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
 
 public class CertificateParser {
 
-    private final CertificateFactory certificateFactory;
-
     public CertificateParser() {
-        certificateFactory = new CertificateFactory();
     }
 
-    public Certificate parse(String certificate) throws IOException, CertificateException {
-        PEMParser pemParser = new PEMParser(new StringReader(certificate));
+    public X509Certificate parse(String certificate) throws IOException, CertificateException {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this method to throw at most one checked exception instead of: java.io.IOException, java.security.cert.CertificateException [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS1160)